diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 8e5e68073..1bff0128a 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -135,7 +135,7 @@ endif (HAVE_ZLIB)
 # We need to link with libatomic on systems that do not have builtin atomics, or
 # don't have builtin support for 8 byte atomics
 set(protobuf_LINK_LIBATOMIC false)
-if (NOT MSVC)
+if (NOT MSVC AND NOT CMAKE_SYSTEM_NAME STREQUAL "IOS")
   include(CheckCXXSourceCompiles)
   set(OLD_CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})
   set(CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS} -std=c++11)
@@ -149,7 +149,7 @@ if (NOT MSVC)
     set(protobuf_LINK_LIBATOMIC true)
   endif (NOT protobuf_HAVE_BUILTIN_ATOMICS)
   set(CMAKE_REQUIRED_FLAGS ${OLD_CMAKE_REQUIRED_FLAGS})
-endif (NOT MSVC)
+endif (NOT MSVC AND NOT CMAKE_SYSTEM_NAME STREQUAL "IOS")
 
 if (protobuf_BUILD_SHARED_LIBS)
   set(protobuf_SHARED_OR_STATIC "SHARED")
